packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart

```diff
@@ -734,6 +729,172 @@ class TimelineGridPainter extends CustomPainter {
+          verticalScrollOffset: verticalScrollOffset,
+          horizontalScrollOffset: horizontalScrollOffset,
+          chartStartInset: chartStartInset,
+        );
+
+  static const strokeWidth = 4.0;
+  static const bracketWidth = 24.0;
+  static const bracketCurveWidth = 8.0;
+  static const bracketVerticalPadding = 8.0;
+
+  final TimelineFrame selectedFrame;
+
+  final int startTimeOffsetMicros;
+
+  final double startingPxPerMicro;
+
+  final double Function(TimelineEvent) yForEvent;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    if (selectedFrame == null) return;
+
+    canvas.clipRect(Rect.fromLTWH(
+      0.0,
+      rowHeight, // We do not want to paint inside the timestamp section.
+      constraints.maxWidth,
+      constraints.maxHeight - rowHeight,
+    ));
+
+    final paint = Paint()
+      ..color = timelineSelectionColor
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = strokeWidth;
+
+    _paintBrackets(canvas, paint, event: selectedFrame.uiEventFlow);
+    _paintBrackets(canvas, paint, event: selectedFrame.rasterEventFlow);
+  }
+
+  void _paintBrackets(
+    Canvas canvas,
+    Paint paint, {
+    @required TimelineEvent event,
+  }) {
+    final startMicros = event.time.start.inMicroseconds - startTimeOffsetMicros;
+    final endMicros = event.time.end.inMicroseconds - startTimeOffsetMicros;
+    final startPx = startMicros * startingPxPerMicro * zoom + chartStartInset;
+    final endPx = endMicros * startingPxPerMicro * zoom + chartStartInset;
+
+    final startBracketX =
+        startPx - visible.left + (bracketWidth - bracketCurveWidth);
+    final endBracketX =
+        endPx - visible.left - (bracketWidth - bracketCurveWidth);
+    final bracketTopY = yForEvent(event) - visible.top - bracketVerticalPadding;
+    final bracketBottomY = bracketTopY +
+        event.depth * rowHeightWithPadding -
+        rowPadding +
+        bracketVerticalPadding * 2;
+
+    // Draw the start bracket.
+    canvas.drawPath(
+      Path()
+        ..moveTo(startBracketX, bracketTopY)
+        ..lineTo(startBracketX - bracketWidth + bracketCurveWidth, bracketTopY)
+        ..arcTo(
+          Rect.fromLTWH(
+            startBracketX - bracketWidth,
+            bracketTopY,
+            bracketCurveWidth * 2,
+            bracketCurveWidth * 2,
+          ),
+          degToRad(270),
+          degToRad(-90),
+          false,
+        )
+        ..lineTo(
+          startBracketX - bracketWidth,
+          bracketBottomY - bracketWidth,
+        )
+        ..arcTo(
+          Rect.fromLTWH(
+            startBracketX - bracketWidth,
+            bracketBottomY - bracketVerticalPadding * 2,
+            bracketCurveWidth * 2,
+            bracketCurveWidth * 2,
+          ),
+          degToRad(180),
+          degToRad(-90),
+          false,
+        )
+        ..lineTo(startBracketX, bracketBottomY),
+      paint,
+    );
+
+    // Draw the end bracket.
+    canvas.drawPath(
```

Comment:
Maybe you can reuse the start bracket code if you flip the end bracket horizontally?

https://stackoverflow.com/a/3129152

Reply:
That a neat idea. Added a todo to implement this.

Reply:
🚀 this is a beautiful PR, ship it!